apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-previewing-theme.ts

```diff
@@ -45,6 +45,13 @@ export const usePreviewingTheme = () => {
 	const previewingThemeSlug = usePreviewingThemeSlug();
 	const { previewingThemeName } = useSelect(
 		( select ) => {
+			// Avoid calling getTheme with null - fetches all site /themes (10s~)
+			if ( ! previewingThemeSlug ) {
+				return {
+					previewingThemeSlug,
+					previewingThemeName: undefined,
+				};
+			}
```

Comment:
Using both approaches to be a little more defensive with this hook, only one is technically needed.